simtools/model/mirrors.py

```diff
@@ -136,23 +181,47 @@ def get_single_mirror_parameters(self, number):
 
         Returns
         -------
-        (pos_x, pos_y, diameter, flen, shape): tuple of float
-            X, Y positions, diameter, focal length and shape.
+        (pos_x, pos_y, mirror_diameter, focal_length, shape_type): tuple of float
+            X, Y positions, mirror_diameter, focal length and shape_type.
         """
 
-        if number > self.number_of_mirrors - 1:
-            self._logger.error("Mirror number is out range")
-            return None
-        return (
-            self._mirrors["pos_x"][number],
-            self._mirrors["pos_y"][number],
-            self._mirrors["diameter"][number],
-            self._mirrors["flen"][number],
-            self._mirrors["shape"][number],
-        )
-
-    def plot_mirror_layout(self):
-        """
-        Plot the mirror layout (not implemented yet).
-        """
-        raise NotImplementedError
+        mask = self.mirror_table["mirror_panel_id"] == number
+        if not np.any(mask):
+            self._logger.debug(f"Mirror id{number} not in table, using first mirror instead")
+            mask[0] = True
+        try:
+            return_values = (
+                u.Quantity(
+                    self.mirror_table[mask]["mirror_x"].value[0],
+                    self.mirror_table[mask]["mirror_x"].unit,
+                ),
+                u.Quantity(
+                    self.mirror_table[mask]["mirror_y"].value[0],
+                    self.mirror_table[mask]["mirror_y"].unit,
+                ),
+                u.Quantity(
```

Comment:
Do you need to explicitly build the Quantity here? If you simply return `self.mirror_table[mask]["mirror_diameter"][0]`, won't you get the quantity already?

Reply:
Yes, this is neccessary: `self.mirror_table[mask]["mirror_diameter"][0]` would only contain the value without unit. `self.mirror_table[mask]["mirror_diameter"]` is a astropy column with Quantity entries and `type(u.Quantity(sim_t[mask]["mirror_diameter"]).value)` is a `numpy.ndarray`.

Reply:
Ahh OK, I see. Thanks!